src/suspendplugin.c

```diff
@@ -118,44 +121,69 @@ handle_nl_command_ack(
 {
     int *ret = arg;
     *ret = 0;
+    DBG("%d", *ret);
     return NL_STOP;
 }
 
 static
 int
-handle_nl_command_family(
+handle_nl_seq_check(
     struct nl_msg *msg,
     void *arg)
 {
-    struct multicast_group *grp = arg;
-    struct nlattr *tb[CTRL_ATTR_MAX + 1];
-    struct genlmsghdr *gnlh = nlmsg_data(nlmsg_hdr(msg));
-    struct nlattr *mcgrp;
-    int rem_mcgrp;
+    DBG("");
 
-    nla_parse(tb, CTRL_ATTR_MAX, genlmsg_attrdata(gnlh, 0), genlmsg_attrlen(gnlh, 0), NULL);
+    return NL_OK;
+}
 
-    if (!tb[CTRL_ATTR_MCAST_GROUPS]) return NL_SKIP;
+static
+gboolean
+suspend_plugin_netlink_handler()
+{
+    struct nl_cb *cb;
+    int res = 0;
+    int err = 0;
 
-    nla_for_each_nested(mcgrp, tb[CTRL_ATTR_MCAST_GROUPS], rem_mcgrp) {
-        struct nlattr *tb_mcgrp[CTRL_ATTR_MCAST_GRP_MAX + 1];
+    cb = nl_cb_alloc(NL_CB_VERBOSE);
+    if (!cb) {
+        connman_error("%s: failed to allocate netlink callbacks", __func__);
+        return FALSE;
+    }
 
-        nla_parse(tb_mcgrp, CTRL_ATTR_MCAST_GRP_MAX, nla_data(mcgrp), nla_len(mcgrp), NULL);
+    err = 1;
+    nl_cb_err(cb, NL_CB_CUSTOM, handle_nl_command_error, &err);
+    nl_cb_set(cb, NL_CB_VALID, NL_CB_CUSTOM, handle_nl_command_valid, &err);
+    nl_cb_set(cb, NL_CB_FINISH, NL_CB_CUSTOM, handle_nl_command_finished, &err);
+    nl_cb_set(cb, NL_CB_ACK, NL_CB_CUSTOM, handle_nl_command_ack, &err);
+    nl_cb_set(cb, NL_CB_SEQ_CHECK, NL_CB_CUSTOM, handle_nl_seq_check, &err);
 
-        if (!tb_mcgrp[CTRL_ATTR_MCAST_GRP_NAME] || !tb_mcgrp[CTRL_ATTR_MCAST_GRP_ID]) {
-            continue;
-        }
+    res = nl_recvmsgs(nl_socket, cb);
+    if (res < 0) {
+        connman_error("nl_recvmsgs failed - wmtWifi %s:%d\n", __func__, res);
+        nl_cb_put(cb);
+        return FALSE;
+    }
 
-        if (strncmp(nla_data(tb_mcgrp[CTRL_ATTR_MCAST_GRP_NAME]), grp->group,
-                nla_len(tb_mcgrp[CTRL_ATTR_MCAST_GRP_NAME]))) {
-            continue;
-        }
+    if (err != 0 && err != 1) {
```

Comment:
This `res` and `err` checking could be simplified and multiple `nl_cb_put()` calls eliminated if you make `suspend_plugin_netlink_handler()` to return `void`. Its return value is not being used anyway.

Reply:
ok